diff -rup lcdproc-0.0.10.sav/lcdproc.c lcdproc-0.0.10/lcdproc.c
--- lcdproc-0.0.10.sav/lcdproc.c	Sun Jan  8 18:28:59 2006
+++ lcdproc-0.0.10/lcdproc.c	Sun Jan  8 20:02:31 2006
@@ -51,8 +51,8 @@ static const char * PrioBackFunctionText
 class cLcdFeed : public cStatus {
 protected:
   virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber);
-  virtual void Recording(const cDevice *Device, const char *Name);
-  virtual void Replaying(const cControl *DvbPlayerControl, const char *Name);
+  virtual void Recording(const cDevice *Device, const char *Name, const char *FileName, bool On);
+  virtual void Replaying(const cControl *DvbPlayerControl, const char *Name, const char *FileName, bool On);
   virtual void SetVolume(int Volume, bool Absolute);
   virtual void OsdClear(void);
   virtual void OsdTitle(const char *Title);
@@ -80,20 +80,20 @@ void cLcdFeed::ChannelSwitch(const cDevi
   }
 }
 
-void cLcdFeed::Recording(const cDevice *Device, const char *Name)
+void cLcdFeed::Recording(const cDevice *Device, const char *Name, const char *FileName, bool On)
 {
   //syslog(LOG_INFO, "lcdproc: cLcdFeed::Recording  %d %s", Device->CardIndex(), Name);
-  if (Name)
+  if (On)
     LCDproc->SetCardStat(Device->CardIndex(),2);
   else
     LCDproc->SetCardStat(Device->CardIndex(),1);	  
 }
 
-void cLcdFeed::Replaying(const cControl *DvbPlayerControl, const char *Name)
+void cLcdFeed::Replaying(const cControl *DvbPlayerControl, const char *Name, const char *FileName, bool On)
 {
   //syslog(LOG_INFO, "lcdproc: cLcdFeed::Replaying %s",  Name);
-  replaymode=(Name)?true:false;
-  if ( replaymode ) {
+  replaymode=On;
+  if (replaymode) {
     LCDproc->SetReplayDevice( (cDvbPlayerControl *) DvbPlayerControl);
     LCDproc->SetMain(2, Name);
     LCDproc->SetThreadState( (cLcd::ThreadStates) 2); // Replaying
